lms/templates/navigation.html

```diff
@@ -21,6 +20,26 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
+% if uses_pattern_library:
+    <%block name="js_extra">
+    <%static:require_module module_name="js/header_factory" class_name="HeaderFactory">
+    HeaderFactory();
+    </%static:require_module>
+    </%block>
+% endif
+
+<%
+if uses_pattern_library:
```

Comment:
This is my one remaining blocker. I spoke with @AlasdairSwan offline about this, but I'm also curious for your perspective, @andy-armstrong.